x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/unified/UnifiedChatCompletionRequestEntity.java

```diff
@@ -0,0 +1,172 @@
+                    if (message.toolCallId() != null) {
+                        builder.field(TOOL_CALL_ID_FIELD, message.toolCallId());
+                    }
+                    if (message.toolCalls() != null) {
+                        builder.startArray(TOOL_CALLS_FIELD);
+                        for (UnifiedCompletionRequest.ToolCall toolCall : message.toolCalls()) {
+                            builder.startObject();
+                            {
+                                builder.field(ID_FIELD, toolCall.id());
+                                builder.startObject(FUNCTION_FIELD);
+                                {
+                                    builder.field(ARGUMENTS_FIELD, toolCall.function().arguments());
+                                    builder.field(NAME_FIELD, toolCall.function().name());
+                                }
+                                builder.endObject();
+                                builder.field(TYPE_FIELD, toolCall.type());
+                            }
+                            builder.endObject();
+                        }
+                        builder.endArray();
+                    }
+                }
+                builder.endObject();
+            }
+        }
+        builder.endArray();
+
+        if (unifiedRequest.maxCompletionTokens() != null) {
+            builder.field(MAX_COMPLETION_TOKENS_FIELD, unifiedRequest.maxCompletionTokens());
+        }
+
+        builder.field(NUMBER_OF_RETURNED_CHOICES_FIELD, 1);
+
+        if (unifiedRequest.stop() != null && unifiedRequest.stop().isEmpty() == false) {
+            builder.field(STOP_FIELD, unifiedRequest.stop());
+        }
+        if (unifiedRequest.temperature() != null) {
+            builder.field(TEMPERATURE_FIELD, unifiedRequest.temperature());
+        }
+        if (unifiedRequest.toolChoice() != null) {
+            if (unifiedRequest.toolChoice() instanceof UnifiedCompletionRequest.ToolChoiceString) {
+                builder.field(TOOL_CHOICE_FIELD, ((UnifiedCompletionRequest.ToolChoiceString) unifiedRequest.toolChoice()).value());
+            } else if (unifiedRequest.toolChoice() instanceof UnifiedCompletionRequest.ToolChoiceObject) {
+                builder.startObject(TOOL_CHOICE_FIELD);
+                {
+                    builder.field(TYPE_FIELD, ((UnifiedCompletionRequest.ToolChoiceObject) unifiedRequest.toolChoice()).type());
+                    builder.startObject(FUNCTION_FIELD);
+                    {
+                        builder.field(
+                            NAME_FIELD,
+                            ((UnifiedCompletionRequest.ToolChoiceObject) unifiedRequest.toolChoice()).function().name()
+                        );
+                    }
+                    builder.endObject();
+                }
+                builder.endObject();
+            }
+        }
+        if (unifiedRequest.tools() != null && unifiedRequest.tools().isEmpty() == false) {
```

Comment:
```suggestion
       boolean usesTools = unifiedRequest.tools() != null && unifiedRequest.tools().isEmpty() == false

        if (usesTools) {
```

Rationale: [Complex conditionals](https://github.com/zakirullin/cognitive-load?tab=readme-ov-file#complex-conditionals); to be fair it's not a very complex conditional, but IMHO it's still easier to read, if it's a "positive" formulated boolean